cmake/SetupCCache.cmake

```diff
@@ -10,9 +10,12 @@ if(USE_CCACHE)
       # In order to make ccache work with precompiled headers we need to:
       # - Hash the header file in the repo that will generate the precompiled
       #   header
-      # - Ignore the precompiled header in the build directory
+      # - In the past we had to ignore the precompiled header in the build
+      #   directory:
+      #   "CCACHE_IGNOREHEADERS=${CMAKE_BINARY_DIR}/SpectrePch.hpp:${CMAKE_BINARY_DIR}/SpectrePch.hpp.gch "
+      #   This doesn't seem to be necessary anymore, but may help if issues
```

Comment:
Is there a reason to remove this? If not, I'd like to keep it as-is. I think actually CI caching was broken completely until these ignores were added. It's been working fine for many years, so why change it?

Reply:
Dropped this change, but removed the `SpectrePch.hpp.gch` part because it doesn't exist anymore. CMake generates the PCH somewhere in `CMakeFiles/`. I'll do some more tests to make sure ccache is working correctly.